shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -240,13 +244,96 @@ bool FlutterWindowsEngine::Run() {
   return Run("");
 }
 
+std::pair<std::string, int> GetReleaseVersionAndBuildNumber() {
+  char modulePath[MAX_PATH];
+  // Get the full path of the currently running executable
+  if (GetModuleFileNameA(NULL, modulePath, MAX_PATH) == -1) {
+    return {"Error retrieving module file name.", -1};
+  }
+
+  // Get the size of the version information
+  DWORD handle = -1;
+  DWORD versionInfoSize = GetFileVersionInfoSizeA(modulePath, &handle);
+  if (versionInfoSize == -1) {
+    return {"Error retrieving version info size.", -1};
+  }
+
+  // Allocate memory for version info
+  std::vector<char> versionData(versionInfoSize);
+  if (!GetFileVersionInfoA(modulePath, handle, versionInfoSize,
+                           versionData.data())) {
+    return {"Error retrieving version info.", -1};
+  }
+
+  // Get the version info structure
+  VS_FIXEDFILEINFO* fileInfo = nullptr;
+  UINT fileInfoSize = -1;
+  if (!VerQueryValueA(versionData.data(), "\\",
+                      reinterpret_cast<LPVOID*>(&fileInfo), &fileInfoSize)) {
+    return {"Error querying version info.", -1};
+  }
+
+  if (fileInfo) {
+    // Extract version numbers
+    DWORD major = HIWORD(fileInfo->dwFileVersionMS);
+    DWORD minor = LOWORD(fileInfo->dwFileVersionMS);
+    DWORD build = HIWORD(fileInfo->dwFileVersionLS);
+
+    char version[49];
+    snprintf(version, sizeof(version), "%lu.%lu.%lu", major, minor, build);
+    return {std::string(version), LOWORD(fileInfo->dwFileVersionLS)};
+  }
+
+  return {"No version information available.", -1};
+}
+
+std::string GetLocalAppDataPath() {
+  PWSTR path = nullptr;
+  HRESULT result = SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, NULL, &path);
+  if (SUCCEEDED(result)) {
+    std::wstring widePath(path);
+    std::string localAppDataPath(widePath.begin(), widePath.end());
+    // The calling process is responsible for freeing this resource
+    // https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath
+    CoTaskMemFree(path);
+    return localAppDataPath;
+  } else {
+    FML_LOG(ERROR) << "Failed to retrieve the local AppData directory.";
+    return "";
+  }
+}
+
 bool FlutterWindowsEngine::Run(std::string_view entrypoint) {
   if (!project_->HasValidPaths()) {
     FML_LOG(ERROR) << "Missing or unresolvable paths to assets.";
     return false;
   }
   std::string assets_path_string = project_->assets_path().u8string();
   std::string icu_path_string = project_->icu_path().u8string();
+  auto shorebird_yaml_path =
+      fml::paths::JoinPaths({assets_path_string, "shorebird.yaml"});
+  std::string* shorebird_yaml_contents = new std::string();
```

Comment:
Does this need to be a pointer? Also it should be held in a unique_ptr if it is a pointer (so that it's deleted at the end of the scope).

Reply:
Nope, it doesn't. Fixed!